times/times.go

```diff
@@ -29,5 +29,9 @@ func TimeToString(time *time.Time) string {
 
 // CompareMessageDates returns false if the dates do not match and the t2 time
 func CompareMessageDates(t1, t2 time.Time) (bool, time.Time) {
-	return t1.Truncate(24 * time.Hour).Equal(t2.Truncate(24 * time.Hour)), t2
+	var match bool
+	if t1.Year() == t2.Year() && t1.YearDay() == t2.YearDay() {
+		match = true
+	}
+	return match, t2
```

Comment:
* You forgot month
* the variable `match` isn't needed, just return directly
* the return value for Time isn't needed either, as you always return t2 anyway

Reply:
* The month does not need to be checked since YearDay returns a day of the year(1-365).
* I removed the `match` variable and added another return instead
* Not sure what you mean here, the t2 is used by `DateDelimiter` function and is therefore returned. I can remove it, but it requires a bit of a rewrite.